frontend/talentsearch/resources/js/components/skills/AddSkillsToExperience/AddSkillsToExperience.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useMemo, useState } from "react";
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+import { Scalars, Skill, SkillFamily } from "@common/api/generated";
+import { Tab, TabSet } from "@common/components/tabs";
+import {
+  FilterIcon,
+  SearchCircleIcon,
+  ChevronDownIcon,
+  ChevronUpIcon,
+} from "@heroicons/react/solid";
+import { matchStringCaseDiacriticInsensitive } from "@common/helpers/formUtils";
+import Pagination, { usePaginationVars } from "@common/components/Pagination";
+import { invertSkillTree } from "@common/helpers/skillUtils";
+import AddedSkills from "../AddedSkills";
+import SkillResults from "../SkillResults";
+import SkillChecklist from "../SkillChecklist";
+import SearchBar from "../SearchBar";
+
+export interface AddSkillsToExperienceProps {
+  allSkills: Skill[];
+  frequentSkills: Skill[];
+  addedSkills: Skill[];
+  onRemoveSkill: (id: Scalars["ID"]) => Promise<void>;
+  onAddSkill: (id: Scalars["ID"]) => Promise<void>;
+}
+
+const AddSkillsToExperience: React.FunctionComponent<
+  AddSkillsToExperienceProps
+> = ({ allSkills, frequentSkills, addedSkills, onRemoveSkill, onAddSkill }) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const [familyFilteredSkills, setFamilyFilteredSkills] = useState<Skill[]>([]);
+  const [searchFilteredSkills, setSearchFilteredSkills] = useState<Skill[]>([]);
+
+  const resultsPaginationPageSize = 5;
+  const frequentSkillsPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    frequentSkills,
+  );
+  const mainstreamSkillsPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    familyFilteredSkills,
+  );
+  const keywordSearchPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    searchFilteredSkills,
+  );
+
+  /**
+   * A handler which takes a list of skill families and filters the allSkills list to
+   * any skills that are a part of those families.  Applies the filter
+   * internally using the useState hook.
+   * @param {SkillFamily[]} checkedFamilies - The selected skill families to filter against.
+   */
+  const handleSkillFamilyChange = (checkedFamilies: SkillFamily[]): void => {
+    const checkedFamilyIds = checkedFamilies.map((family) => family.id);
+    const matchingSkills = allSkills.filter((skill) =>
+      // https://stackoverflow.com/a/39893636
+      skill.families?.some((skillFamily) =>
+        checkedFamilyIds.includes(skillFamily.id),
+      ),
+    );
+    setFamilyFilteredSkills(matchingSkills);
+    mainstreamSkillsPagination.setCurrentPage(1);
```

Comment:
this makes the user lose their current page every time they add/remove a skill and have them go to the first page. Is this the requirement?

Reply:
oh nvm .. I see your comment below in the code :)